code/game/objects/structures/watercloset.dm

```diff
@@ -106,13 +105,18 @@
 		return ..()
 
 /obj/structure/toilet/secret
-	w_items = 2
+	var/obj/item/secret
+	var/secret_type = null
+
+/obj/structure/toilet/secret/Initialize(mapload)
+	. = ..()
+	if (secret_type)
+		secret = new secret_type(src)
+		secret.desc += " It's a secret!"
+		w_items += secret.w_class
+		contents += secret
```

Comment:
Manually setting contents is not needed here. Doing `new secret_type(src)` already makes sure it's inside the toilet.